ffi/src/lib.rs

```diff
@@ -247,17 +255,737 @@ pub extern "C" fn store_open(uri: *const c_char) -> *mut Store {
 
 // TODO: begin_read
 
-// TODO: begin_transaction
+/// Starts a new transaction to allow multiple transacts to be
+/// performed together. This is more efficient than performing
+/// a large set of individual commits.
+///
+/// Returns as [Result<TxReport>](mentat::TxReport) as an [ExternResult](ExternResult).
+///
+/// # Safety
+///
+/// Callers must ensure that the pointer to the [Store](mentat::Store) is not dangling.
+///
+/// Callers are responsible for managing the memory for the return value.
+/// A destructor `tx_report_destroy` is provided for releasing the memory for this
+/// pointer type.
+///
+/// TODO: Document the errors that can result from begin_transaction
+#[no_mangle]
+pub unsafe extern "C" fn store_begin_transaction(store: *mut Store) -> *mut ExternResult {
+    let store = &mut*store;
```

Comment:
Is this spacing our standard?  I would not write `&mutfoo` (it doesn't parse!) so I would expect `&mut *store`.